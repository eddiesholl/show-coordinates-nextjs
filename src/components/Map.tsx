import MapWrapper, { Marker } from "react-map-gl";
import "mapbox-gl/dist/mapbox-gl.css";

const token =
  "pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4M29iazA2Z2gycXA4N2pmbDZmangifQ.-g_vE53SD2WrJ6tFX7QHmA";

const locations = {
  features: [
    {
      type: "Feature",
      properties: {
        title: "Lincoln Park",
        description: "A northside park that is home to the Lincoln Park Zoo",
      },
      geometry: {
        coordinates: [-87.637596, 41.940403],
        type: "Point",
      },
    },
  ],
};
const Map = () => {
  return (
    <MapWrapper
      initialViewState={{
        longitude: -100,
        latitude: 40,
        zoom: 3.5,
      }}
      style={{ width: 600, height: 400 }}
      mapStyle="mapbox://styles/mapbox/streets-v12"
      mapboxAccessToken={token}
    >
      <Marker longitude={-122.4} latitude={37.8} color="red" />
    </MapWrapper>
  );
};

export default Map;
