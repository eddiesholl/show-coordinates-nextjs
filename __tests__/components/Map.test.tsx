import Map from "@/components/Map";
import { Map as MockMap } from "mapbox-gl";

import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import fetchMock from "jest-fetch-mock";

describe("Map", () => {
  beforeEach(() => {
    fetchMock.resetMocks();
    const response = {
      coordinates: [
        { lat: 1, long: 2 },
        { lat: 3, long: 4 },
      ],
    };
    fetchMock.once(JSON.stringify(response));
  });
  it("renders a map", async () => {
    render(<Map />);
    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(MockMap).toHaveBeenCalledTimes(1));
    expect(MockMap).toHaveBeenCalledWith(
      expect.objectContaining({
        initialViewState: {
          bounds: [1, 2, 3, 4],
          fitBoundsOptions: { padding: 40 },
        },
      })
    );
  });

  it("matches the current snapshot", () => {
    const { container } = render(<Map />);
    expect(container).toMatchSnapshot();
  });
});
