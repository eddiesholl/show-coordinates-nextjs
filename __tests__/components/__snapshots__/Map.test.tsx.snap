// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map matches the current snapshot 1`] = `
<div>
  <div
    style="position: relative; width: 600px; height: 400px;"
  />
</div>
`;
